test/checkpoint/test_checkpoint.c

```diff
@@ -217,6 +273,21 @@ main(int argc, char *argv[])
               g.cookies[i].uri, sizeof(g.cookies[i].uri), "%s%04d", URI_BASE, g.cookies[i].id));
         }
 
+        /*
+         * Setup thread data. There are N worker threads, a checkpoint thread and possibly a clock
```

Comment:
More of a question rather than a comment but this comment seems to suggest we may not have a clock thread. In what scenarios do we not have a clock thread? Who will manage the timestamps if we don't have a clock thread?

Reply:
Ah, I added an if condition - we only have the clock thread when -X is passed to the test, to run the test with timestamps. Otherwise timestamps are not used in this test.